File: monitoring/src/main/java/com/example/monitoring/user/adapter/port/in/UserRestController.java
package com.example.monitoring.user.adapter.port.in;

import com.example.monitoring.user.adapter.port.in.request.CreateUserRequest;
import com.example.monitoring.user.adapter.port.in.response.UserResponse;
import com.example.monitoring.user.application.port.in.UserUseCase;
import com.example.monitoring.user.application.port.in.response.UserResponseDto;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

@RestController
@RequiredArgsConstructor
public class UserRestController {

    private final UserUseCase userUseCase;

    @GetMapping("/api/users/{userId}")
    @ResponseStatus(HttpStatus.OK)
    public UserResponse findById(@PathVariable Long userId) {
        UserResponseDto userResponseDto = userUseCase.findById(userId);
        return UserResponse.from(userResponseDto);
    }

    @PostMapping("/api/users")
    @ResponseStatus(HttpStatus.CREATED)
    public UserResponse save(@RequestBody CreateUserRequest createUserRequest) {
        UserResponseDto responseDto = userUseCase.createUser(createUserRequest.toRequestDto());
        return UserResponse.from(responseDto);
    }
}

File: kotlin-spring-boot/src/main/kotlin/com/hello/kotlin/user/presentation/UserRestController.kt
package com.hello.kotlin.user.presentation

import com.hello.kotlin.user.application.UserUseCase
import com.hello.kotlin.user.presentation.request.UserRequest
import com.hello.kotlin.user.presentation.response.UserResponse
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/users")
class UserRestController(
    val userUseCase: UserUseCase
) {

    @GetMapping("/{userId}")
    fun getUser(@PathVariable userId: Long) = UserResponse.from(
        userUseCase.findById(userId)
    )

    @PostMapping("")
    fun createUser(@RequestBody userRequest: UserRequest) = UserResponse.from(
        userUseCase.save(userRequest.toRequestDto())
    )

    @PutMapping("/{userId}")
    fun updateUser(@PathVariable userId: Long,
                   @RequestBody userRequest: UserRequest) = UserResponse.from(
        userUseCase.updateUser(userId, userRequest.toRequestDto())
    )
}